src/app/zap-templates/common/ClustersHelper.js

```diff
@@ -262,7 +262,11 @@ function handleBasic(item, [ atomics, enums, bitmaps, structs ])
     item.name         = item.name || item.label;
     item.isStruct     = false;
     item.atomicTypeId = atomic.atomicId;
-    item.discrete     = atomic.isDiscrete;
+    item.isDiscrete   = atomic.isDiscrete;
+    item.isString     = atomic.isString;
```

Comment:
There are a few rules that seem to add additional constraints for type safety:
  - "signed" has no meaning for string
  - an item cannot be at the same time long, char and/or string
  
 Any other rules?
 
 Could we design a type system that encodes this (e.g. have 'item.type' being STRING or LONG or CHAR)? Maybe instead of LONG it should be INT or NUMERIC (since long seems to imply a size already).
 
 Generally adding extra bools does not seem very easy to maintain.
 
 I am concerned that our code generation increases a lot in complexity and javascript is extra hard to maintain because missing types.

Reply:
I will try to just use `isAnalog`by updating the types definition, so less `bool`s. 

About types in particular, this JS object is just a mirror of what is inside the database. I would expect the parser that ingest the xml definition to complains is there is this kind of errors.

The reason to expose properties in this way is mostly because from the template itself, you can just do `{{#if isDiscrete}}` without having to resort to a javascript function for example.

Also, the specific type of a attributes and command arguments is already carried by the `chipType` property.